libraries/AP_BattMonitor/AP_BattMonitor_Backend.cpp

```diff
@@ -30,15 +30,16 @@ AP_BattMonitor_Backend::AP_BattMonitor_Backend(AP_BattMonitor &mon, AP_BattMonit
 {
 }
 
-/// capacity_remaining_pct - returns the % battery capacity remaining (0 ~ 100)
-uint8_t AP_BattMonitor_Backend::capacity_remaining_pct() const
+/// capacity_remaining_pct - returns true if the battery % is available and writes to the percentage argument
+bool AP_BattMonitor_Backend::capacity_remaining_pct(int8_t &percentage) const
 {
-    float mah_remaining = _params._pack_capacity - _state.consumed_mah;
-    if ( _params._pack_capacity > 10 ) { // a very very small battery
```

Comment:
Why was this logic changed?

It's an odd construct - was there a reason for it in the first place?

Reply:
It was introduced [here](https://github.com/ArduPilot/ardupilot/commit/90378b21dabd4cfc7893e5ae67a1c6ca7e81e246) to avoid dividing by zero when BATT_CAPACITY is set to zero. 

But instead of checking for 0, it used 10 instead. Pretty odd, but harmless I guess.

Reply:
Reverted to original logic